api/users/models.py

```diff
@@ -310,6 +311,9 @@ def remove_as_group_admin(self, group_id: int):
         ).update(group_admin=False)
 
 
+assert issubclass(FFAdminUser, UserABC)
```

Comment:
@matthewelwell this looks a bit ugly, we can't really inherit from the UserABC because of metaclass conflict

Reply:
I don't really follow what this is doing. In fact, I'm surprised it's not raising an AssertionError. If `FFAdminUser` doesn't inherit from `UserABC` this should raise an assertion error as far as I'm aware. 

The following snippet (which seems to be a simple representation of what you guys have here) will raise an AssertionError on the last line, so I'm not sure how your code isn't 🤔 

```python
class Base(ABC):
    @abstractmethod
    def foo(self):
        raise NotImplementedError()
    
class Sub(Base):
    def foo(self):
        return "sub-foo"
    
class NotSub:
    def foo(self):
        return "not-sub-foo"
        
assert issubclass(Sub, Base)  # this is fine
assert issubclass(NotSub, Base)  # this is not!
```

Reply:
That is because of this:  `def __subclasshook__(cls, subclass):`

i.e: this code will work 


```python
from abc import ABC, abstractmethod

class Base(ABC):
    @abstractmethod
    def foo(self):
        raise NotImplementedError()

    @classmethod
    def __subclasshook__(cls, subclass):
        return "foo" in subclass.__dict__

class Sub(Base):
    def foo(self):
        return "sub-foo"

class NotSub:
    def foo(self):
        return "not-sub-foo"

assert issubclass(Sub, Base)  # this is fine
assert issubclass(NotSub, Base)  # this is not!



```